common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -1109,6 +1141,16 @@ private KeyPair generateNewRsaKeyPair(@NonNull final Context context,
         );
     }
 
+    private static boolean isStrongBoxUnavailableException(@NonNull final Throwable t) {
+        final boolean isStrongBoxException = t.getClass().getSimpleName().equals(STRONG_BOX_UNAVAILABLE_EXCEPTION);
```

Comment:
Can we do an instance check here isntead?

Reply:
We cannot due to wackiness discovered by @AdamBJohnsonx - documented via comment here:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java#L1063-L1065

Reply:
Basically - hard refs to the `StrongBoxUnavailableException` class can cause issues on certain devices, so this workaround is being used